library/summary.mli

```diff
@@ -108,5 +108,49 @@ module Interp : sig
 
 end
 
+(** Observables
+
+    [OBSERVABLE] captures the pattern of backtrackable state that can be enabled
+    and disabled. To use it, [register] the value that you want to record and then
+    [activate] and [deactivate] the value using the returned [token].
+*)
+module type OBSERVABLE =
+sig
+  (** The type of tokens to manipulate values *)
+  type token
+
+  (** The value being stored. *)
+  type value
+
+  (** Register a new value and get the token used to enable and disable it. *)
+  val register : name:string -> ?override:bool -> value -> token
+
+  (** Activate/deactive the value attached to the token. *)
+  val activate : token -> unit
+  val deactivate : token -> unit
+end
+
+(** The implementation side of observation.
+    This should be held internally with the creator of the state.
+    Only the [OBSERVABLE] signature should be exposed.
+ *)
+module type OBSERVABLE_USER =
+sig
+  include OBSERVABLE
+
+  (** Determine if the value for the given token is active *)
+  val is_active : token -> bool
+
+  (** Get all of the active values *)
+  val all_active : unit -> (string * value) list
+end
+
+(** Generic implementation of [OBSERVABLE_USER]. *)
+module Make
```

Comment:
probably shouldn't be called just Make (ends up as `Summary.Make` which seems wrong)